shmem.c

```diff
@@ -347,7 +347,12 @@ int shmget(key_t key, size_t size, int flags)
 				path_buffer[path_length] = '\0';
 				int shmid = atoi(path_buffer);
 				if (shmid != 0) {
-					int idx = ashv_read_remote_segment(shmid);
+					int idx = ashv_find_local_index(shmid);
+
+					if (idx == -1) {
```

Comment:
With this if statement in place the bottom condition can be eliminated by using an `else` clause

Reply:
If you turned the next `if` into an `else`, then it would only unlock and return if it found the segment locally. If it didn't find it locally, it would read the remote segment and then skip the return.

Reply:
I don't understand, how are these not identical?
```c
if (idx == -1) {
  idx = ashv_read_remote_segment(shmid);
}

if (idx != -1) {
  pthread_mutex_unlock(&mutex);
  return shmem[idx].id;
}
```

And the patch i'm suggesting:
```c
if (idx == -1) {
  idx = ashv_read_remote_segment(shmid);
} else {
  pthread_mutex_unlock(&mutex);
  return shmem[idx].id;
}
```
You are just comparing `idx` twice, one for equal -1 and one for not equal -1, if it's not equal to -1 then it's not so run the else statement no need to check twice if it's not -1

Reply:
Oh i didn't see you re-assigning `idx` inside that if block. Nevermind